internal/services/subscription/data_source_locations.go

```diff
@@ -0,0 +1,175 @@
+	return "azurerm_locations"
+}
+
+func (r ManagerSubscriptionLocationsDataSource) ModelObject() interface{} {
+	return &ManagerSubscriptionLocationsDataSourceModel{}
+}
+
+func (r ManagerSubscriptionLocationsDataSource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"location": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+		},
+	}
+}
+
+func (r ManagerSubscriptionLocationsDataSource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"region_type": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+		"display_name": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+		"zone_ids": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
+		},
+		"zone_names": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
+		},
+		"zone_mappings": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"logical_zone": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+					"physical_zone": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+				},
+			},
+		},
+	}
+}
+
+func (r ManagerSubscriptionLocationsDataSource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func:    dataSourceLocationsReadFunc,
+	}
+}
+
+func dataSourceLocationsReadFunc(ctx context.Context, metadata sdk.ResourceMetaData) error {
+	client := metadata.Client.Subscription.ResourceManagerClient
+	subscriptionId := metadata.Client.Account.SubscriptionId
+
+	id := commonids.NewSubscriptionID(subscriptionId)
+	resp, err := client.ListLocations(ctx, id, resourcesSubscription.DefaultListLocationsOperationOptions())
+	if err != nil {
+		if resp.HttpResponse.StatusCode == http.StatusNotFound {
+			return fmt.Errorf("%s was not found", id)
+		}
+
+		return fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+
+	var model ManagerSubscriptionLocationsDataSourceModel
+	if err := metadata.Decode(&model); err != nil {
+		return fmt.Errorf("decoding: %+v", err)
+	}
+
+	normalizedLocation := location.Normalize(model.Location)
+	subscriptionLocation := parse.NewSubscriptionLocation(id.SubscriptionId, normalizedLocation)
+
+	locationValue, err := filterLocation(normalizedLocation, resp.Model.Value)
+	if err != nil {
+		return err
+	}
+
+	state := getZonesMapping(locationValue)
+	state.RegionType = string(*locationValue.Metadata.RegionType)
+	state.DisplayName = *locationValue.DisplayName
+	state.Location = normalizedLocation
+
+	metadata.SetID(subscriptionLocation)
+
+	return metadata.Encode(&state)
+}
+
+func filterLocation(location string, input *[]resourcesSubscription.Location) (*resourcesSubscription.Location, error) {
+	for _, item := range *input {
```

Comment:
what if `input == nil` ?

Reply:
Added a check to the azure client response